File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        // int i=0;
        // int j =0;
        // for(i=0;i<n;i++){
        //     for(j = i+1;j<n;j++){
        //         if((nums[i]+nums[j])==target){
        //             return {i,j};
        //         }
                
        //     }
        // }
        // return {i,j};
        map<int,int> mpp;
        for(int i=0;i<n;i++){
            int a = nums[i];
            int m = target-nums[i];
            if(mpp.find(m)!=mpp.end()){
                return {mpp[m],i};
            }
            mpp[a]=i;

        }

        return {-1,-1};
        
        
    }
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int n = nums.length;
        // for(int i=0;i<n;i++){
        //     for(int j=i+1;j<n;j++){
        //         if(nums[i]+nums[j]==target){
        //             return new int[] {i,j};
        //         }
        //     }
        // }
        Map<Integer,Integer> mpp = new HashMap<>();
        for(int i=0;i<n;i++){
            int num = nums[i];
            int moreNeeded = target - num;
            if(mpp.containsKey(moreNeeded)){
                return new int[]{i,mpp.get(moreNeeded)};
            }
            mpp.put(nums[i],i);
        }
        return new int[]{-1,-1};
        
    }
}